### PR TITLE
Fixes MySQLToS3 float to int conversion

### PR DESCRIPTION
**Pull request by gmontanola**
The conversion wasn't being applied because the `astype` method needs to be used in a `pd.Series` object. This means all integer columns that have **NULL** values will be parsed as floats and the resulting file will have float formated numbers.

Also, the Int64Dtype must be passed as an **instance** in order for working as intended.

---

### CODE REVIEW

File: airflow/providers/amazon/aws/transfers/mysql_to_s3.py [hunk 1]
```diff
@@ -105,7 +105,8 @@ def _fix_int_dtypes(self, df):
                 notna_series = df[col].dropna().values
                 if np.isclose(notna_series, notna_series.astype(int)).all():
                     # set to dtype that retains integers and supports NaNs
-                    df[col] = np.where(df[col].isnull(), None, df[col]).astype(pd.Int64Dtype)
+                    df[col] = np.where(df[col].isnull(), None, df[col])
+                    df[col] = df[col].astype(pd.Int64Dtype())
```
Comment (feluelle) [inline airflow/providers/amazon/aws/transfers/mysql_to_s3.py:115]: Don't we have a test for that? Maybe can you add one to see how it should look like after this?

Comment (gmontanola) [reply airflow/providers/amazon/aws/transfers/mysql_to_s3.py:115]: The test for this operator **doesn't** cover this method (`_fix_int_dtypes`). So I'll add one.

Thanks for the input!

Comment (HaloKo4) [reply airflow/providers/amazon/aws/transfers/mysql_to_s3.py:115]: @gmontanola are you planning to finish it?

Comment (gmontanola) [reply airflow/providers/amazon/aws/transfers/mysql_to_s3.py:115]: I totally forgot this. Well, I guess adding this test should be pretty straight forward. I'll try to add this this week. Are you willing to add this by yourself or found a better a way to fix the problem?

**Comment by gmontanola** [comment]
@feluelle 

I added a simple test to check the **integer conversion**.
I'm open to _feedback_ or improvements. I'm really not used to unit testing (but you love to improve :blue_heart:)

image

**Review by feluelle — APPROVED** [review]
LGTM

**Review by potiuk — APPROVED** [review]

**Pull request merged**